subprojects/core/src/main/java/org/gradle/configuration/project/DefaultCompileOperationFactory.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.configuration.project;
+
+import org.gradle.api.internal.DocumentationRegistry;
+import org.gradle.api.internal.cache.StringInterner;
+import org.gradle.configuration.CompileOperationFactory;
+import org.gradle.configuration.ScriptTarget;
+import org.gradle.groovy.scripts.ScriptSource;
+import org.gradle.groovy.scripts.internal.BuildScriptData;
+import org.gradle.groovy.scripts.internal.BuildScriptDataSerializer;
+import org.gradle.groovy.scripts.internal.BuildScriptTransformer;
+import org.gradle.groovy.scripts.internal.CompileOperation;
+import org.gradle.groovy.scripts.internal.FactoryBackedCompileOperation;
+import org.gradle.groovy.scripts.internal.InitialPassStatementTransformer;
+import org.gradle.groovy.scripts.internal.NoDataCompileOperation;
+import org.gradle.groovy.scripts.internal.SubsetScriptTransformer;
+
+public class DefaultCompileOperationFactory implements CompileOperationFactory {
+    private static final StringInterner INTERNER = new StringInterner();
+    private static final String CLASSPATH_COMPILE_STAGE = "CLASSPATH";
+    private static final String BODY_COMPILE_STAGE = "BODY";
```

Comment:
💄 That will be the 5th place in the code base where we have a representation for that exact same piece of knowledge, maybe it's time to unify?

Reply:
Not sure I understood. I am aware of this being used here: https://github.com/gradle/gradle/pull/12655/files#diff-78678a11af673b66fd983e90cbc14218L54-L56
and this class extracted that piece of knowledge so that it can be used in another place.

Reply:
I mean, `"BODY"` and `"CLASSPATH"` are effectively API and have multiple declarations within the system, we could maybe create a single home for this script related build operations and reuse across Groovy and Kotlin.